convert_hf_to_gguf.py

```diff
@@ -1024,6 +1071,101 @@ def _set_vocab_builtin(self, model_name: Literal["gpt-neox", "llama-spm"], vocab
             self.gguf_writer.add_add_eos_token(field.parts[-1].tolist()[0])
 
 
+# for converting mmproj file
+class MultimodalModel:
+    hparams: dict
+    dir_model: Path
+    ftype: gguf.LlamaFileType
+    fname_out: Path
+    tensor_map: gguf.TensorNameMap
+    gguf_writer: gguf.GGUFWriter
+
+    def __init__(self, hparams: dict, ftype: gguf.LlamaFileType, fname_out: Path, endianess: gguf.GGUFEndian, use_temp_file: bool):
+        self.hparams = hparams
+        self.ftype = ftype
+        self.fname_out = fname_out
+        self.tensor_map = gguf.get_tensor_name_map(gguf.MODEL_ARCH.CLIP_VISION, 128)
+        self.gguf_writer = gguf.GGUFWriter(path=None,
+                                           arch="clip",
+                                           endianess=endianess,
+                                           use_temp_file=use_temp_file)
+
+    def set_gguf_parameters(self, n_embd_text: int):
+        """Function to be called by Model.set_gguf_parameters()"""
+        self.gguf_writer.add_type(gguf.GGUFType.CLIP_VISION)
+        self.gguf_writer.add_file_type(self.ftype)
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.PROJECTION_DIM, n_embd_text)
+        self.gguf_writer.add_bool(gguf.Keys.ClipVision.HAS_VISION_ENCODER, True)
+
+        # vision config
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.IMAGE_SIZE,           self.find_hparam(["image_size"]))
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.PATCH_SIZE,           self.find_hparam(["patch_size"]))
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.EMBEDDING_LENGTH,     self.find_hparam(["hidden_size"]))
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.FEED_FORWARD_LENGTH,  self.find_hparam(["intermediate_size"]))
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.BLOCK_COUNT,          self.find_hparam(["num_hidden_layers"]))
+        self.gguf_writer.add_uint32(gguf.Keys.ClipVision.Attention.HEAD_COUNT, self.find_hparam(["num_attention_heads"]))
```

Comment:
Note that I didn't add `gguf_writer.add_*` pattern because I don't yet have the full list of keys. This can be done in a follow-up PR